packages/compass-connections/src/stores/connections-store.ts

```diff
@@ -272,7 +272,10 @@ export function useConnections({
   recentConnections: ConnectionInfo[];
   favoriteConnections: ConnectionInfo[];
   cancelConnectionAttempt: (connectionInfoId: string) => Promise<void>;
-  connect: (connectionInfo: ConnectionInfo) => Promise<void>;
```

Comment:
it's a bit unsettling that TS didn't complain about the actual implementation having two parameters, just because the second was optional 🤔

Reply:
That's kind of just how TS works, a function that takes 1 required parameter + 1 optional parameter can be interacted with in the same way as a function that only takes 1 required parameter, so there's no way for TS to tell that this wasn't intentional

Reply:
Yes I get the reasoning. I was wondering if there is some other way this could've been defined so that the mistake would've been caught earlier. But I guess the only way would've been to make the parameter required, which doesn't make sense in this use case.